doc/custom-configuration.md

```diff
@@ -91,3 +91,38 @@ spec:
 
 Cassandra will load the `cassandra.yaml.d/100-concurrent.yaml` file as well as the default settings managed by the operator!
 
+### Prometheus support
+
+The Cassandra image that is included with this project has the [cassandra-exporter](https://github.com/instaclustr/cassandra-exporter) built in, so it is ready to export the metrics to Prometheus instance. In order to use this capability automatically, one must have a running Prometheus operator in the same Kubernetes environment and have Prometheus object following a set of ServiceMonitors via a selector (a set of labels); for example:
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  name: cassandra
+spec:
+  serviceAccountName: prometheus
+  serviceMonitorSelector:
+    matchLabels:
+      prometheus-server: main
+  replicas: 1
+  version: v1.7.1
+  resources:
+    requests:
+      memory: 400Mi
+```
+
+`prometheus` service account used above is taken literally from the Prometheus' [user guide](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#enable-rbac-rules-for-prometheus-pods). Use the example above to create Prometheus object that uses a specified set of ServiceMonitor labels.
+
+  When Prometheus object is created, and the labels are at hand, add the following configuration to the cluster CRD:
+```yaml
+...
+spec:
+ ...
+  prometheusServiceMonitorLabels:
```

Comment:
I disagree with this. As mentioned in my [comment](https://github.com/instaclustr/cassandra-operator/pull/294#discussion_r328405381), these labels _are_ specific to the Prometheus support.  These labels come from the Prometheus' selector for the ServiceMonitors, they are needed so that Prometheus picks up the ServiceMonitor _we_ create. These specific labels have nothing to do with our internal services, which we create for the Cassandra discovery. We only use them all the way Prometheus->ServiceMonitor->Service because it's convenient and easy to debug if there are issues. We could have a constant label such as `prometheus: <cdc name>` to uniquely pick the service from the ServiceMonitor, without asking users for labels, if we didn't have to integrate with Prometheus itself.

That being said, I can definitely see a benefit in the ability to add arbitrary labels to any node, service, StatefulSet or pod created by the operator. In theory, it could overlap/encompass this feature by providing a more generic labelling mechanism. I've created a separate ticket #299 for that, as it would be way more expansive than just for this feature.

Reply:
What I meant was that we don't need a special field for "Prometheus-related labels", we can achieve the same end result with a generic service labels field since in any case this is up to the user to figure out what kind of selector (if any) they have in their Prometheus config and apply matching labels to the service. But anyway, feel free to disregard this suggestion :-)

Reply:
:+1:

At the moment, until we implement some kind of "generic" labelling, I prefer keeping the name specific to Prometheus, as that would be more clear to the users...IMHO, of course.